server/vulnerabilities/nvd/cpe_test.go

```diff
@@ -266,11 +327,22 @@ func TestTranslateSoftwareToCPE(t *testing.T) {
 				Version:          "0.3",
 				BundleIdentifier: "vendor2",
 				Source:           "apps",
+				GenerateCPE:      "something_wrong",
+			},
+			// This 'duplicated' entry if for testing that 'UpsertSoftwareCPEs' is only called iff
+			// software.GenerateCPE != detected CPE
+			{
+				ID:               3,
+				Name:             "Product2",
+				Version:          "0.3",
+				BundleIdentifier: "vendor2",
+				Source:           "apps",
+				GenerateCPE:      "cpe:2.3:a:vendor2:product4:0.3:*:*:*:*:macos:*:*",
```

Comment:
Where is this entry duplicated? (does it match an entry in `XmlCPETestDict`?)

Reply:
Both softwares (ID:2 and ID3) will match to the same CPE, the difference is that for Software{ID:2} -> `GenerateCPE != cpe` and for Software{ID:3} -> `GenerateCPE == cpe`. By adding Software{ID:3} I'm testing that the `UpsertSoftwareCPEs` method is only going to be called because of Software{ID:2} - otherwise [this assertion ](https://github.com/fleetdm/fleet/pull/10514/files/d65518655c4eef845e677163612208b3b0d5bd1e#diff-4140fbb3f7fb5b9cf64d161b23f690a3e916744f6fa7148ff7f19a4827f130dbR359) would fail (because `UpsertSoftwareCPEs` would be called twice with `cpe:2.3:a:vendor2:product4:0.3:*:*:*:*:macos:*:*`)

Hopefully that makes sense :P

Reply:
```
  <cpe-item name="cpe:/a:vendor2:product4:999">
    <title xml:lang="en-US">Vendor2 Product4 999 for MacOS</title>
    <cpe-23:cpe23-item name="cpe:2.3:a:vendor2:product4:999:*:*:*:*:macos:*:*"/>
  </cpe-item>
```
Sorry, I'm confused with `999` vs `0.3`, the CPE file says `<cpe-item name="cpe:/a:vendor2:product4:999">` and the generated CPE says `vendor2:product4:0.3`.

Reply:
We DM'd and Juan clarified that we match vendor and product_name and the version is set from the software piece.